proposed/http-client/http-client.md

```diff
@@ -3,16 +3,31 @@ HTTP Client
 
 This document describes common interfaces for sending HTTP messages.
 
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 
 ## Specification
 
+### Client
+
 An HTTP client has the responsibility to send a PSR-7 request and return a PSR-7
-response. When there is an error during sending the request or an error with network an
-exception should be thrown. 
+response. The HTTP client MAY modify the request, given by the user and sent to 
+the server or the response from the server before returning. In this case the 
+request and the response must be consistent between the body and headers. I.e. a 
+server may return a gzip encoded body and the client may know how to decode this, 
+when it decodes the body the client MUST also remove the header that specifies this 
+encoding.
+
+### Exceptions
 
 An implementing library MUST implement `Psr\Http\Client\ClientException` for each exception it throws. 
-It SHOULD implement exceptions for `Psr\Http\Client\Exception\NetworkException` and
-`Psr\Http\Client\Exception\RequestException`.
+
+When there is an error with the network or the remote server cannot be reached, the
+HTTP client SHOULD thrown a `Psr\Http\Client\Exception\NetworkException`. If the 
```

Comment:
`thrown` -> `throw`

Generally as a rule, `SHOULD` is recommended against as it cannot be relied upon by a client anymore. Is there rationale as to why this isn't a `MUST`?

Reply:
in some circumstances it might be difficult to know what kind of error happened. there will probably always be some fuzzyness around the exception type. however, we are specific as to which things cause a network exception, so i think MUST should be fine here.

Reply:
can you fix the thrown throw please?

ok if we say MUST here?